torch/distributed/_functional_collectives.py

```diff
@@ -616,3 +621,24 @@ def reduce_scatter_tensor_inplace(
     legacy_allgather: all_gather_tensor_inplace,
     legacy_reducescatter: reduce_scatter_tensor_inplace,
 }
+
+def _all_gather_into_tensor_coalesced_fallback(output_tensors, input_tensors, group, async_op=False):
```

Comment:
I assume we should remove this "fallback" api once the _coalescing_manager adopted?

Reply:
Well, I still want to support GLOO, so I'll need a version of this somewhere.
The NCCL path will use _coalescing_manager when it adds support for all_gather.